### PR TITLE
Team 2 UI Feature and Team 2/5 In-Game build menu collab

### PR DESCRIPTION
**Pull request by The-AhmadAA**
This PR brings in the implementation of In-game UI elements, and the In-Game Tower build menu
Specifically:
* Addition of a Pause button to the main screen that pauses the game and brings up a Pause menu with the appropriate buttons
* Addition of a wave timer and mob count to the main game screen that shows the remaining mobs in the current wave
* Addition of level background music and ambient sounds
* Addition of the In-Game build menu which uses the towers selected by the user in the Turret Select screen, and allows spawning of towers into the map, provided the tile is not occupied and the resource cost can be covered.
* Other UI improvements such as some button sounds, build sounds and a Skin file configured for scene2d.ui

There are additional improvements/integration possible, but these are dependent on other features. Please feel free to ask questions or provide feedback as necessary.

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -381,6 +374,26 @@ private Entity spawnPlayer(GridPoint2 position) {
     return newPlayer;
   }
 
+  // commented 383 - 386 out as there was a missing arg?
+//  private void spawnDemonBoss() {
+//    Entity demon = MobBossFactory.createDemonBoss();
+//    spawnEntityAt(demon, new GridPoint2(19, 5), true, false);
+//  }
+
+  private void spawnPatrick() {
```
Comment (meganroxburgh) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:383]: I think this mob spawn has been removed, as it's now covered in the switch statement of spawnMob. @samsully can you confirm?

**Review by meganroxburgh — APPROVED** [review]
All looks good - just the one comment about the spawnPatrick but I'm not 100% certain. Not a major issue though, its easy to remove/add back in if necessary :)

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java (hunk 1 above)
Comment (The-AhmadAA) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:383]: Yeah ForestGameArea is constantly changing, we may need to deal with this in the next sprint. We haven't modified any of the mob spawning in this file, but have removed any hardcoded tower spawning as it's not necessary now.

**Comment by ryandmcneilly** [comment]
Hey just some small issues I noticed, when the player picks less than 5 turrets the game crashes (`IndexOutOfBounds`) however when you pick 0 zero turrets the game still works (just noticed this is a known issue on discord). Also when you try and select multiple towers it seems the buttons stay highlighted causing some bugs. Apart from that looks good!

**Comment by The-AhmadAA** [comment]
> Hey just some small issues I noticed, when the player picks less than 5 turrets the game crashes (`IndexOutOfBounds`) however when you pick 0 zero turrets the game still works (just noticed this is a known issue on discord). Also when you try and select multiple towers it seems the buttons stay highlighted causing some bugs. Apart from that looks good!

Hey Ryan, thanks for the feedback, this all seems to be related to Turret Select Screen, so will need to consult with the author. It currently doesn't affect gameplay if all or no towers are selected, we can probably deal with this in a Bug report ticket

**Comment by The-AhmadAA** [comment]
This PR now also includes updating of buttons globally using the newly built Skin "glass.json"

**Pull request merged**